### PR TITLE
Update readthedocs with hivemind.optim

### PR DESCRIPTION
**Pull request by yhn112**

### CODE REVIEW

**Comment by yhn112** [comment]
Any ideas for improving the documentation are welcome

**Review by justheuristic — APPROVED** [review]
I would recommend merging this as is and thinking about next steps in a separate thread :)

File: docs/modules/optim.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+**hivemind.optim**
+==================
+
+.. automodule:: hivemind.optim
+.. currentmodule:: hivemind.optim
+
```
Comment (justheuristic) [inline docs/modules/optim.rst:6]:
```suggestion

.. raw:: html

  This module contains decentralized optimizers that wrap regular pytorch optimizers to collaboratively train a shared model. Depending on the specific type, optimizers may average model parameters with peers, exchange gradients, or follow a more complicated distributed training strategy.
  <br><br>
```

File: docs/modules/index.rst [hunk 2]
```diff
@@ -5,7 +5,8 @@
...
    :maxdepth: 2
 
+   optim
    averaging
+   dht
    client
-   server
-   dht
\ No newline at end of file
+   server
```
Comment (mryab) [inline docs/modules/index.rst]: Maybe sort alphabetically?

**Review by mryab — APPROVED** [review]

File: docs/modules/index.rst (hunk 2 above)
Comment (yhn112) [reply docs/modules/index.rst]: Nope.
At least, I would like to see client and server next to each other.
Also, for now, optim and averaging seem most useful for an average user, thus we begin with them.

**Pull request merged**